src/circuit/zsetoperator/filter.rs

```diff
@@ -0,0 +1,71 @@
+//! Defines an operator whose input is a stream of z-sets and which applies
```

Comment:
This operator could be much more generic.  You only need these trait bounds on elements of the input stream:
```
Z: IntoIterator<Item=(T,W)>,
&Z: IntoIterator<Item=&(T,W)>,
```
and the output stream (which does not have to be the same ZSet type as input):
```
O: FromIterator<Item=(T,W)>
```

Unless we expect that Z sets can be more efficiently filtered without converting them into an iterator.